x64_dbg_gui/Project/Src/Memory/MemoryPage.cpp

```diff
@@ -1,47 +1,44 @@
 #include "MemoryPage.h"
 
-MemoryPage::MemoryPage(uint_t parBase, uint_t parSize, QObject* parent) : QObject(parent)
+MemoryPage::MemoryPage(uint_t parBase, uint_t parSize, QObject* parent) : QObject(parent), mBase(0), mSize(0)
 {
     Q_UNUSED(parBase);
     Q_UNUSED(parSize);
-    mBase = 0;
-    mSize = 0;
 }
 
-bool MemoryPage::read(void* parDest, uint_t parRVA, uint_t parSize)
+bool MemoryPage::read(void* parDest, uint_t parRVA, uint_t parSize) const
 {
-    return DbgMemRead(mBase + parRVA, (unsigned char*)parDest, parSize);
+    return DbgMemRead(mBase + parRVA, reinterpret_cast<unsigned char*>(parDest), parSize);
 }
 
-bool MemoryPage::read(byte_t* parDest, uint_t parRVA, uint_t parSize)
+bool MemoryPage::read(byte_t* parDest, uint_t parRVA, uint_t parSize) const
 {
-    return read((void*)parDest, parRVA, parSize);
+    return read(reinterpret_cast<void*>(parDest), parRVA, parSize);
 }
 
-bool MemoryPage::write(const void* parDest, uint_t parRVA, uint_t parSize)
+bool MemoryPage::write(const void* parDest, uint_t parRVA, uint_t parSize) const
 {
-    bool ret = DbgFunctions()->MemPatch(mBase + parRVA, (unsigned char*)parDest, parSize);
+    bool ret = DbgFunctions()->MemPatch(mBase + parRVA, reinterpret_cast<const unsigned char*>(parDest), parSize);
     GuiUpdatePatches();
     return ret;
 }
 
-bool MemoryPage::write(const byte_t* parDest, uint_t parRVA, uint_t parSize)
+bool MemoryPage::write(const byte_t* parDest, uint_t parRVA, uint_t parSize) const

```

Comment:
True! But because the underlying code doesn't modify any MemoryPage class members, it can still be set to const. 

Having the write() function as const or not in this instance is more of a design choice, I guess. It's up to you whether you think having this a non-const makes more sense from an interface perspective.

I'm happy to remove it from write() if you would prefer.